File: client/store-main/src/app/service/user.service.ts
import { Injectable } from '@angular/core';
import User from '../Components/Entity/User';
import { HttpClient } from '@angular/common/http';

const BASE_URL_FOR_REGISTER = "http://localhost:8084/user"
const BASE_URL_FOR_Login = "http://localhost:8084/user/login/"
const BASE_URL_FOR_GETTINGALL ="http://localhost:8084/user/";

@Injectable({
  providedIn: 'root'
})
export class UserService {

  createUser(user: User) {
    return this.http.post(BASE_URL_FOR_REGISTER, user);
  }

  login(userName, password) {
    return this.http.get(BASE_URL_FOR_Login + userName + "/" + password);
  }

  getAllUser(){
    return this.http.get<User[]>(BASE_URL_FOR_GETTINGALL);
  }

  constructor(private http: HttpClient) { }
}

File: server/NgoLogin/src/main/java/com/NgoLogin/service/UserService.java
package com.NgoLogin.service;

import java.util.List;

import com.NgoLogin.entity.User;

public interface UserService {

	public User addUser(User user);

	public List<User> getUsers();

	public List<User> login(String username, String password);

	public boolean deleteUser(Long uId);

	public User updateUser(Long uId, User user);

}
